### PR TITLE
TCP echo example

### PR DESCRIPTION
**Pull request by MingweiSamuel**

### CODE REVIEW

**Comment by MingweiSamuel** [comment]
Issue was caused by the fact that `.lines()` strips the newline characters at the end, so they all became one big line when echoed back. Have to manually re-append the newline characters

File: hydroflow/tests/surface_network.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+        Ok(()) as Result<(), Box<dyn Error>>
+    });
+
+    local.await;
+
+    Ok(())
+}
+
+#[test]
```
Comment (jhellerstein) [inline hydroflow/tests/surface_network.rs:75]: Should this still be copied into here?

File: hydroflow_lang/src/graph/ops/mod.rs [hunk 2]
```diff
@@ -368,6 +368,33 @@ pub const OPERATORS: [OperatorConstraints; 19] = [
             }
         }),
     },
+    OperatorConstraints {
```
Comment (jhellerstein) [inline hydroflow_lang/src/graph/ops/mod.rs:371]: Do you want this hack checked into the repo? Would be better not to, IMHO

**Review by jhellerstein — APPROVED** [review]

**Comment by MingweiSamuel** [comment]
```rs
    /// Server
    member_reqs = members_in -> tee();
    cross_join = join();
    member_reqs[0] -> foreach(|m| send(m, "ack")}
    member_reqs[1] -> map(|r| ((), r)) -> [0]cross_join;
    msgs_in -> map(|m| ((), m)) -> [1]cross_join;
    cross_join -> foreach(|(_, (msg, member))| send(member, msg));

    /// Client
    recv_iter(vec![0]) -> foreach(|_x|, send(server_port, "{}:{}", local_ip, CLIENT_PORT));
```

**Comment by MingweiSamuel** [comment]
#192

**Pull request closed without merging**